daemon/mgr/container_storage.go

```diff
@@ -471,96 +471,60 @@ func (mgr *ContainerManager) setMountTab(ctx context.Context, c *Container) erro
 	return nil
 }
 
-func (mgr *ContainerManager) setRootfsQuota(ctx context.Context, c *Container) error {
-	logrus.Debugf("start to set rootfs quota, dir(%s)", c.MountFS)
+func (mgr *ContainerManager) setDiskQuota(ctx context.Context, c *Container, mounted bool) error {
+	var globalQuotaID uint32
 
-	if c.MountFS == "" {
-		return nil
-	}
-
-	rootfsQuota := quota.GetDefaultQuota(c.Config.DiskQuota)
-	if rootfsQuota == "" {
-		return nil
-	}
-
-	qid := "0"
-	if c.Config.QuotaID != "" {
-		qid = c.Config.QuotaID
-	}
-
-	id, err := strconv.Atoi(qid)
-	if err != nil {
-		return errors.Wrapf(err, "failed to change quota id(%s) from string to int", qid)
-	}
-
-	// set rootfs quota
-	_, err = quota.SetRootfsDiskQuota(c.MountFS, rootfsQuota, uint32(id))
-	if err != nil {
-		return errors.Wrapf(err, "failed to set rootfs quota, mountfs(%s), quota(%s), quota id(%d)",
-			c.MountFS, rootfsQuota, id)
-	}
-
-	return nil
-}
-
-func (mgr *ContainerManager) setMountPointDiskQuota(ctx context.Context, c *Container) error {
-	if c.Config.DiskQuota == nil {
-		if c.Config.QuotaID != "" && c.Config.QuotaID != "0" {
-			return fmt.Errorf("invalid argument, set quota-id without disk-quota")
-		}
-		return nil
-	}
-
-	var (
-		qid        uint32
-		setQuotaID bool
-	)
-
-	if c.Config.QuotaID != "" {
+	if c.Config.QuotaID != "" && c.Config.QuotaID != "0" {
```

Comment:
maybe we need lock here? This action will take long time. If we don't add lock, there maybe some confusing with concurrent update request

Reply:
update and create should be conflict, so I think no need to add lock here